src/plugins/data/common/search/utils.ts

```diff
@@ -14,14 +14,18 @@ import type { IKibanaSearchResponse } from './types';
  * @returns true if response had an error while executing in ES
  */
 export const isErrorResponse = (response?: IKibanaSearchResponse) => {
-  return !response || !response.rawResponse || (!response.isRunning && !!response.isPartial);
+  return (
+    !response ||
+    !response.rawResponse ||
+    (!response.isRunning && !!response.isPartial && !response.rawResponse?._clusters?.details)
```

Comment:
Just for reference:

This basically determines that ES will handle the `isPartial` flag differently for CCS?

Reply:
Yeah, I should add a comment there. Basically, the behavior for CCS with `ccs_minimize_roundtrips=true` is to set `is_partial` to `true` if the search is complete but there are some shard failures (and thus partial results), then have details about the failures inside the `_clusters.details` section. This will also likely be the behavior of CCS with `ccs_minimize_roundtrips=false` and non-CCS after https://github.com/elastic/elasticsearch/pull/98913 is resolved.

Reply:
I've added some comments to the code in 0859897808dedf4581ddff1227181ad6c9c0ed35.

Reply:
> This will also likely be the behavior of CCS with ccs_minimize_roundtrips=false